main/src/ca/uwaterloo/flix/language/phase/unification/EffUnification3.scala

```diff
@@ -416,7 +416,7 @@ object EffUnification3 {
     */
   def simplify(tpe: Type): Type = try {
     // Check if type is too complex to simplify via Zhegalkin polynomials.
-    if (tpe.typeVars.size > SetUnification.MaxVars) {
+    if (tpe.typeVars.size > SetUnification.MaxVars / 2) {
```

Comment:
Why is this not reverted?

Reply:
Rerunning the solver on the minimized formulas throw stack overflow if this is larger. again note that this only touches type simplification, not regular solving